valohai_yaml/objs/pipelines/execution_node.py

```diff
@@ -3,42 +3,55 @@
 from valohai_yaml.lint import LintResult
 from valohai_yaml.objs.pipelines.node import ErrorAction, Node
 from valohai_yaml.objs.pipelines.node_action import NodeAction
-from valohai_yaml.types import LintContext, NodeOverrideDict
+from valohai_yaml.objs.pipelines.override import Override
+from valohai_yaml.types import LintContext, SerializedDict
 
 
 class ExecutionNode(Node):
     """Represents an execution node within a pipeline definition."""
 
     type = 'execution'
-
+    override: Override
     def __init__(
         self,
         *,
         name: str,
         step: str,
         actions: Optional[List[NodeAction]] = None,
-        override: Optional[NodeOverrideDict] = None,
+        override: Optional[SerializedDict] = None,
```

Comment:
Why the type change? If anything, this should allow an `Override` object too now that we have one.

```suggestion
        override: Optional[Union[Override, NodeOverrideDict]] = None,
```

Reply:
Override.parse as a classmethod works with type Dict[str, Any] so Union no need imho

Reply:
The point is you should be able to pass in an Override you've built by hand if you're using this as a programmatic API (like e.g. https://github.com/valohai/papi does).